### ISSUE TITLE
Mixing OpenTracing and automatic instrumentation

### ISSUE THREAD

**Issue by daniel-white** [DataDog/dd-trace-dotnet#508]
I'm using 1.7 of the .NET stuff for .NET Core. I followed the example on how to add OpenTracing  to my DI and sprinkled it thru my code. (https://docs.datadoghq.com/tracing/advanced/opentracing/?tab=net)

```csharp
public static IServiceCollection AddTracer(this IServiceCollection services)
{
  ITracer tracer = OpenTracingTracerFactory.CreateTracer();
  GlobalTracer.Register(tracer);
  services.TryAddSingleton(tracer);
  return services;
 }
```

```csharp
using (IScope scope = _tracer.BuildSpan(someSpan).StartActive())
{
 // logic
}
```

 However, I'm not seeing the the custom traces/spans on the flame charts. I am seeing the automatically instrumented spans. 

Is there a special sauce I'm missing? Does this require a certain version of the agent to be run?

(edit: formatting)

**Comment by colin-higgins** [comment]
Hi @daniel-white 
Current behavior is that the OpenTracing traces and automatic Traces are not submitted in the same trace, so they will appear under different traces in the UI.
This is because have totally independent instances of Tracers. 
Uniting these when they are both present is on our roadmap, and I will update you with more information on this.
The current behavior in the back-end is also that the top level spans for any given service must have the same `ResourceName`. Only the top level traces which correspond to the primary resource name (which would presumably be the one the automatic traces are creating) may show.

If you would like to be sure that your traces all show up together in the meantime, using the static tracer reference is an option.

**Comment by daniel-white** [comment]
@colin-higgins: thanks! i was able to hack to construct the internal type of the `OpenTracingTracer` providing the static DD tracer. Now they are showing up. Perhaps it would be good to add documentation around this limitation. i look forward to when it is properly integrated (soon i hope).

**Comment by alexeyzimarev** [comment]
@colin-higgins the issue is that the Datadog tracer is very limited. I cannot find any way to access the scope context and extract/inject it for distributed tracing. OpenTracing has it all. Without distributed tracing, the whole tracing idea is almost pointless.

**Comment by zacharycmontoya** [comment]
Hi @daniel-white, we've just released version 1.8.0 of the tracer which adds a new API `OpenTracingTracerFactory.WrapTracer` (thank you @alexeyzimarev for the contribution!). Using this for your use-case should look like the following:

```csharp
public static IServiceCollection AddTracer(this IServiceCollection services)
{
  ITracer tracer = OpenTracingTracerFactory.WrapTracer(Datadog.Trace.Tracer.Instance);
  GlobalTracer.Register(tracer);
  services.TryAddSingleton(tracer);
  return services;
}
```

Please let us know if you run into any other issues and feedback, as we're looking to better support manual instrumentation and OpenTracing scenarios.

**Comment by lucaspimentel** [comment]
Looks like this issue is resolved with [release 1.8.0](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.8.0), so I'm closing this issue.

**Comment by sabbadino** [comment]
You should update the documentation here 
https://docs.datadoghq.com/tracing/opentracing/dotnet/
